src/mathematics/glyphinfo.rs

```diff
@@ -0,0 +1,142 @@
+use truetype::{Result, Tape, Value};
+
+use layout::Coverage;
+use super::ValueRecord;
+
+table! {
+    @position
+    #[doc = "A table of positioning information defined on a per-glyph bases."]
+    pub GlyphInfo { // MathGlyphInfo
+        italics_correction_offset      (u16), // MathItalicsCorrectionInfo
+        top_accent_offset              (u16), // MathTopAccentAttachment
+        extended_shape_coverage_offset (u16), // ExtendedShapeCoverage
+        kern_info_offset               (u16), // MathKernInfo
+
+        italics_corrections (Option<ItalicsCorrections>) |this, tape, position| {
+            jump_take_maybe!(tape, position, this.italics_correction_offset)        
+        },
+
+        top_accent_attachments (Option<TopAccentAttachments>) |this, tape, position| {
+            jump_take_maybe!(tape, position, this.top_accent_offset)
+        },
+
+        extended_shape_coverage (Option<Coverage>) |this, tape, position| {
+            jump_take_maybe!(tape, position, this.extended_shape_coverage_offset)
+        },
+
+        kern_info (Option<KernInfo>) |this, tape, position| {
+            jump_take_maybe!(tape, position, this.kern_info_offset)
+        },
+    }
+}
+
+table! {
+    @position
+    #[doc = "A table of italics corrections."]
+    pub ItalicsCorrections { // MathItalicsCorrectionInfo
+        coverage_offset (u16), // Coverage
+        count           (u16), // ItalicsCorrectionCount
+
+        corrections (Vec<ValueRecord>) |this, tape, _| { // ItalicsCorrection
+           tape.take_given(this.count as usize)
+        },
+
+        coverage (Coverage) |this, tape, position| {
+            jump_take!(tape, position, this.coverage_offset)
+        },
+    }
+}
+
+table! {
+    @position
+    #[doc = "A table of horizontal positioning for top accents."]
+    pub TopAccentAttachments { // MathTopAccentAttachment
+        coverage_offset (u16), // TopAccentCoverage
+        count           (u16), // TopAccentAttachmentCount
+
+        attachments (Vec<ValueRecord>) |this, tape, _| {    
+           tape.take_given(this.count as usize)
+        },
+
+        coverage (Coverage) |this, tape, position| {
+            jump_take!(tape, position, this.coverage_offset)
+        },
+    }
+}
+
+table! {
+    @position
+    #[doc = "A table of kerning information for mathematical glyphs."]
+    pub KernInfo {
+        coverage_offset (u16), // MathKernCoverage
+        count           (u16), // MathKernCount
+
+        kernings (Vec<KernGlyph>) |this, tape, position| {
+            let mut kern_glyphs: Vec<KernGlyph> = Vec::with_capacity(this.count as usize);
```

Comment:
I try to choose as neutral names as possible inside such reading blocks. Let’s rename `kern_glyphs` to `values` and `kern` to `value`.